### PR TITLE
Bump Rubocop to 0.54.x

### LINKED ISSUE #6655: Update dependency constraint to allow for rubocop v0.52.1
Hey there! :wave:

I noticed that the constraint you have for rubocop doesn't allow for the latest version to be used.

The constraint I found was `~> 0.51.0`, and the latest version available is `0.52.1`.

Can you look into updating that constraint so our users can use the latest and greatest version? Thanks! :revolving_hearts:

### PR DESCRIPTION
**Pull request by crispgm**
- [x] Update Gemfile
- [x] Fix cops

Fix #6655 #6830 #6865 
Closes #6626

### CODE REVIEW

**Comment by DirtyF** [comment]
@crispgm Do you intend to fix remaining offenses?

**Comment by crispgm** [comment]
@DirtyF I may have a try. And I wonder why there are even more issues using 0.51.0.

**Comment by crispgm** [comment]
@DirtyF there are a lot of empty comments in the codebase. Should remove them or add some simple comment?

**Comment by ashmaroli** [comment]
> there are a lot of empty comments in the codebase. Should I disable it?

disable the cop only for `/features/`

**Comment by crispgm** [comment]
@DirtyF Finally, it's happy again! And it's time to make maintainers happy. :)

File: lib/jekyll/drops/site_drop.rb [hunk 1]
```diff
@@ -25,6 +25,7 @@ def key?(key)
         (@obj.collections.key?(key) && key != "posts") || super
       end
 
+      # rubocop:disable Naming/MemoizedInstanceVariableName
```
Comment (ashmaroli) [inline lib/jekyll/drops/site_drop.rb]: Should we just disable this cop globally?

File: test/test_site.rb [hunk 2]
```diff
@@ -213,6 +213,7 @@ def generate(site)
       @site.process
       # exclude files in symlinked directories here and insert them in the
       # following step when not on Windows.
+      # rubocop:disable Lint/NestedPercentLiteral
```
Comment (ashmaroli) [inline test/test_site.rb]: IMO this comment ought to be placed at the top level (just before `should "sort pages alphabetically" do`) and the enabler-comment at the corresponding end..

Comment (crispgm) [reply test/test_site.rb]: Are there any best practice on this? Here I disable the minimum scope for the cops.

Comment (DirtyF) [reply test/test_site.rb]: Ideally we don't add any rubocop comments and lists files to exclude for now in the rubocop config file and anyone can pick a file and fix it like we did before.

Comment (crispgm) [reply test/test_site.rb]: @Di r

**Comment by DirtyF** [comment]
@jekyllbot: :shipit: +dev

**Pull request merged**